pkg/util/selector.go

```diff
@@ -171,3 +162,118 @@ func extractClusterFields(cluster *clusterv1alpha1.Cluster) labels.Set {
 
 	return clusterFieldsMap
 }
+
+// GeneralPolicy is an abstract object of ClusterOverridePolicy, OverridePolicy, PropagationPolicy and ClusterPropagationPolicy.
+type GeneralPolicy interface {
+	// GetName returns the name of Policy
+	GetName() string
+	// GetNamespace returns the namespace of Policy
+	GetNamespace() string
+	// GetResourceSelectors returns the ResourceSelector of Policy
+	GetResourceSelectors() []policyv1alpha1.ResourceSelector
+}
+
+type orderPolicy struct {
+	policyName             string
+	policyNamespace        string
+	policyResourceSelector policyv1alpha1.ResourceSelector
+}
+
+// SortPolicy returns a GeneralPolicy slice sorted by implicit priority.
+func SortPolicy(policies []GeneralPolicy, matchingResourceSelector func(GeneralPolicy) []policyv1alpha1.ResourceSelector) []GeneralPolicy {
```

Comment:
This algorithm looks a bit complicated, can you remind me of its benefits?

Reply:
There are two benefits.

First, it uniformly implements the implicit priority sorting of pp and op (the original sorting algorithm is very unusual).

Second, when the explicit priority is introduced in the future, it will be expanded better.

Reply:
> the original sorting algorithm is very unusual

Do you mean the algorithm introduced by #2267 ?

Reply:
No, I mean, the following algorithm is unusual.
https://github.com/karmada-io/karmada/blob/9bd626b3cc11a4dfc1856182b5cbd648a05184d0/pkg/detector/detector.go#L426-L433